src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -289,42 +349,105 @@ public boolean isPrivateIpUsed() {
         return privateIpUsed;
     }
 
+    @DataBoundSetter
+    public void setPrivateIpUsed(boolean privateIpUsed) {
+        this.privateIpUsed = privateIpUsed;
+    }
+
     public boolean isAlwaysReconnect() {
         return alwaysReconnect;
     }
 
+    @DataBoundSetter
+    public void setAlwaysReconnect(boolean alwaysReconnect) {
+        this.alwaysReconnect = alwaysReconnect;
+    }
+
     public String getLabelString() {
         return labelString;
     }
 
     public int getIdleMinutes() {
-        return (idleMinutes != null) ? idleMinutes : 0;
+        return idleMinutes;
+    }
+
+    @DataBoundSetter
+    public void setIdleMinutes(int idleMinutes) {
+        this.idleMinutes = Math.max(0, idleMinutes);
     }
 
     public int getMaxSize() {
         return maxSize;
     }
 
+    @DataBoundSetter
+    public void setMaxSize(int maxSize) {
+        if (maxSize < minSize) {
+            int newMaxSize = Math.max(1, minSize);
+            warning("Cloud parameter 'maxSize' can't be less than 'minSize' or 1, setting to %d", newMaxSize);
+            maxSize = newMaxSize;
+        }
+        this.maxSize = Math.max(1, maxSize);
+    }
+
     public int getMinSize() {
         return minSize;
     }
 
-    public int getMinSpareSize() {
+    @DataBoundSetter
+    public void setMinSize(int minSize) {
+        if (minSize < 0) {
+            warning("Cloud parameter 'minSize' can't be less than 0, setting to 0");
+        }
+        minSize = Math.max(0, minSize);
+        //TODO: This validation is only in place for unit tests since constructor is run twice on CasC load but not
+        // for unit tests
+        if (minSize > maxSize) {
+            warning("Cloud parameter 'minSize' cannot be greater than 'maxSize', setting 'maxSize' to %d. " +
+                    "Ignore this if caused after a CasC load.", minSize);
+            this.maxSize = minSize;
```

Comment:
This is weird but I guess we don't have an option until we find an alternative. :-#